main.go

```diff
@@ -263,29 +263,31 @@ How to install go: https://golang.org/doc/install
 	if tag == "devel" {
 		commitAndDate := rawVersion[tagIdx+1:]
 		// Remove commit hash and architecture from version
-		date := commitAndDate[strings.IndexByte(commitAndDate, ' ')+1 : strings.LastIndexByte(commitAndDate, ' ')]
+		startDateIdx := strings.IndexByte(commitAndDate, ' ') + 1
+		endDateIdx := strings.LastIndexByte(commitAndDate, ' ')
+		if endDateIdx <= 0 {
+			return errors.New("Can't recognize devel build timestamp")
+		}
+		date := commitAndDate[startDateIdx:endDateIdx]
 
 		versionDate, err := time.Parse(gitTimeFormat, date)
 		if err != nil {
-			fmt.Printf("Can't recognize devel build timestamp: %v\n", err)
-			return true
+			return fmt.Errorf("Can't recognize devel build timestamp: %v", err)
 		}
 
 		if versionDate.After(minGoVersionDate) {
-			return true
+			return nil
 		}
 
-		fmt.Printf("You use the old unstable %q Go version, please upgrade Go to %s\n", rawVersion, supportedGoVersions)
-		return false
+		return fmt.Errorf("You use the old unstable \"%s\" Go version, please upgrade Go to %s", rawVersion, supportedGoVersions)
```

Comment:
you reverted the %q change here, which was good.

also, I would keep the Printfs. errors in Go shouldn't be capitalized or be multiline. if you're printing many lines of output, just print them, don't return an error. We don't really need an error anyway, because we just print it.

Reply:
And when you undo the error return, undo the func rename as well

Reply:
I wanted to reuse the error output mechanism that is in main1 and end the process with an error.

Reply:
Sure. It's still pretty non-standard for errors to be capitalized or have newlines, though, so this will mean warnings from tools like staticcheck. I'm not saying it's wrong, but it's going to require going against standard tools, I think.

Reply:
I suggest doing os.Exit(1) after checking, and output the errors themselves simply via fmt.

Reply:
Yep, that works.